64_MergeSort/app_resources/merge_sort.hlsl

```diff
@@ -0,0 +1,77 @@
+#pragma wave shader_stage(compute)
+
+[[vk::binding(0,0)]] RWStructuredBuffer<int> output_buffer;
+[[vk::binding(1,0)]] RWStructuredBuffer<int> input_buffer;
+
+struct SortingPhaseData
+{
+    uint num_elements_per_array;
+    uint buffer_length;
+};
+
+// Rather than indexing into the input array directly, use group shared memory.
+groupshared int shared_memory_input_array[MaxNumberOfArrayElementsSharedMemoryCanHold];
```

Comment:
I'd make a separate HLSL header like counting sort to shove all the logic into, also use the Accessor Pattern to make the algorithm independent of the resource thats providing the inputs/outputs to the algo

Reply:
Also you know, using a template let us sort something other than `int` (as long as it has an `operator<`